### PR TITLE
Task 1 pathing - module integration

### PR DESCRIPTION
**Pull request by jhuang38**

### CODE REVIEW

File: path_2023.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+from modules import upload_commands
+from modules import waypoint_names_to_coordinates
+from modules import waypoints_to_commands
+from modules import add_takeoff_and_landing_command
+
+
+WAYPOINT_FILE_PATH = pathlib.Path(".", "waypoints", "wrestrc_waypoints.csv")
+CAMERA = 0
+ALTITUDE = 40
+CONNECTION_ADDRESS = "tcp:localhost:5763"
```
Comment (jhuang38) [inline path_2023.py:20]: Obviously change this for local setup

**Comment by jhuang38** [comment]
Note: on my end, it seems that main runs to completion, but I don't see anything different on the Mission Planner GUI (i.e. no commands being uploaded/waypoints placed on the map) - perhaps we can try running this on the UWARG computer and see if we get better results.

File: path_2023.py (hunk 1 above)
Comment (Xierumeng) [reply path_2023.py:20]: Change to `"tcp:localhost:14550"`

File: path_2023.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+"""
+Task 1 path.
+"""
+import pathlib
+import dronekit
```
Comment (Xierumeng) [inline path_2023.py:5]: Nit: Single empty line above this.

File: path_2023.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+from modules import load_waypoint_name_to_coordinates_map
+from modules import qr_input
+from modules import qr_to_waypoint_names
+from modules import upload_commands
+from modules import waypoint_names_to_coordinates
+from modules import waypoints_to_commands
+from modules import add_takeoff_and_landing_command
```
Comment (Xierumeng) [inline path_2023.py:13]: Nit: Sort alphabetically.

File: path_2023.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+
+WAYPOINT_FILE_PATH = pathlib.Path(".", "waypoints", "wrestrc_waypoints.csv")
+CAMERA = 0
+ALTITUDE = 40
+CONNECTION_ADDRESS = "tcp:localhost:14550"
+drone = dronekit.connect(CONNECTION_ADDRESS, wait_ready = True)
```
Comment (Xierumeng) [inline path_2023.py:20]: Move this under `run()`

File: path_2023.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        print("ERROR: qr_to_waypoint_names")
+        return -1
+
+    result, waypoints = waypoint_names_to_coordinates.waypoint_names_to_coordinates(waypoint_names, waypoint_name_to_coordinates)
+    if not result:
+        print("ERROR: waypoint_names_to_coordinates")
+        return -1
+    
+    waypoint_commands = waypoints_to_commands.waypoints_to_commands(waypoints, ALTITUDE)
+
```
Comment (Xierumeng) [inline path_2023.py:44]: Nit: Remove this empty line.

File: path_2023.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    if len(waypoint_commands) == 0:
+        print("Error: waypoints_to_commands")
+        return -1
+    
+    takeoff_landing_commands = add_takeoff_and_landing_command.add_takeoff_and_landing_command(waypoint_commands, ALTITUDE)
+    if len(takeoff_landing_commands) == 0:
+        print("Error: add_takeoff_and_landing_command")
+        return -1
+    
+    upload_commands.upload_commands(drone, takeoff_landing_commands)
```
Comment (Xierumeng) [inline path_2023.py:54]: This function returns a status, please check it.

File: path_2023.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    
+
```
Comment (Xierumeng) [inline path_2023.py:56]: Nit: Remove this empty line.

File: path_2023.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    return 0
+
+
+if __name__ == "__main__":
+    result_run = run()
+    if result_run < 0:
+        print("ERROR")
+    print("Done")
```
Comment (Xierumeng) [inline path_2023.py:64]: Nit: Add single empty line at end of file.

**Comment by Xierumeng** [comment]
Rename PR title to something like `Task 1 pathing`

File: path_2023.py (hunk 2 above)
Comment (Xierumeng) [reply path_2023.py:6]: What I meant is empty line between pathlib and dronekit imports. No empty line between triple quotation marks and `import pathlib` .

File: path_2023.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+from modules import upload_commands
+from modules import waypoint_names_to_coordinates
+from modules import waypoints_to_commands
+
+
+WAYPOINT_FILE_PATH = pathlib.Path(".", "waypoints", "wrestrc_waypoints.csv")
+CAMERA = 0
+ALTITUDE = 40
+CONNECTION_ADDRESS = "tcp:localhost:14550"
+
```
Comment (Xierumeng) [inline path_2023.py:21]: Nit: 2 empty lines between constants and top level functions.

**Review by Xierumeng — APPROVED** [review]

**Pull request merged**